source/MRViewer/MRRenderPointsObject.cpp

```diff
@@ -171,6 +172,65 @@ void RenderPointsObject::forceBindAll()
     bindPoints_();
 }
 
+RenderBufferRef<Vector3f> RenderPointsObject::loadVertPosBuffer_()
+{
+    auto& glBuffer = GLStaticHolder::getStaticGLBuffer();
+    if ( !( dirty_ & DIRTY_POSITION ) || !objPoints_->pointCloud() )
+        return glBuffer.prepareBuffer<Vector3f>( vertPosSize_, false );
+
+    const auto& points = objPoints_->pointCloud()->points;
+    const auto num = objPoints_->pointCloud()->validPoints.find_last() + 1;
+    const auto step = objPoints_->getRenderDiscretization();
```

Comment:
optimization for `step == 1`
```suggestion
    auto& glBuffer = GLStaticHolder::getStaticGLBuffer();
    if ( !( dirty_ & DIRTY_POSITION ) || !objPoints_->pointCloud() )
        return glBuffer.prepareBuffer<Vector3f>( vertPosSize_, false );

    const auto step = objPoints_->getRenderDiscretization();
    const auto& points = objPoints_->pointCloud()->points;
    const auto num = objPoints_->pointCloud()->validPoints.find_last() + 1;
    if ( step == 1 )
        return RenderBufferRef<Vector3f>( points.vec_.data(), vertPosSize_ = num, true );